### PR TITLE
Allow firebase staging DB access + better parser and firebase configuration on localhost + display parser/db url on homepage

### LINKED ISSUE #1322: Set up new firebase instance, use it for localhost+staging, and use existing one for production only

### PR DESCRIPTION
**Pull request by AlexNisnevich**
Fix #1322, #1346, #1347.

### CODE REVIEW

File: src/common/constants.ts [hunk 1]
```diff
@@ -108,10 +143,31 @@ export function stringToType(str: string): CardType {
...
+const STAGING_PARSER_URL = '//parser-staging.wordbots.io';
+const LOCAL_PARSER_URL = 'http://localhost:8080';
+
+export const PARSER_URL: string = (() => {
+  // tslint:disable-next-line: no-typeof-undefined
+  if (typeof window !== 'undefined') {
+    if (['app.wordbots.io', 'wordbots-game.herokuapp.com'].includes(window.location.hostname)) {
+      // On app.wordbots.io or wordbots-game-staging.herokuapp.com, use production DB
+      return LIVE_PARSER_URL;
+    } else if (window.location.hostname === 'localhost') {
```
Comment (jacobnisnevich) [inline src/common/constants.ts:156]: Any reason not to do this as 
```
if (localhost) {
 // local/staging parser URL logic
} else {
  // production
}
```

File: src/common/constants.ts [hunk 2]
```diff
@@ -108,10 +143,31 @@ export function stringToType(str: string): CardType {
...
-const liveParserUrl = '//parser.wordbots.io';
-const stagingParserUrl = '//wordbots-parser-staging.herokuapp.com';
-export const PARSER_URL = USE_STAGING_PARSER ? stagingParserUrl : (shouldUseLocalParser ? `http://localhost:${LOCAL_PARSER_PORT}` : liveParserUrl);
+const LIVE_PARSER_URL = '//parser.wordbots.io';
+const STAGING_PARSER_URL = '//parser-staging.wordbots.io';
+const LOCAL_PARSER_URL = 'http://localhost:8080';
+
+export const PARSER_URL: string = (() => {
+  // tslint:disable-next-line: no-typeof-undefined
+  if (typeof window !== 'undefined') {
```
Comment (jacobnisnevich) [inline src/common/constants.ts:152]: Why is this necessary? If its checking if its running in browser don't we have a helper function for that?

File: webpack.config.js [hunk 3]
```diff
@@ -59,9 +59,7 @@ const webpackConfig = {
     new CopyWebpackPlugin([{from: 'static'}]),
     new webpack.IgnorePlugin(/canvas/),
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),  // Ignore all locale files of moment.js
-    new webpack.DefinePlugin({
-      'process.env': { NODE_ENV: JSON.stringify(NODE_ENV) }
-    })
+    new webpack.EnvironmentPlugin({ NODE_ENV: 'development', PARSER: null, FIREBASE_DB: null })
```
Comment (jacobnisnevich) [inline webpack.config.js:62]: Is there any way to start wordbots locally and overrides these currently?

**Review by jacobnisnevich — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 3 above)
Comment (AlexNisnevich) [reply webpack.config.js:62]: Yeah, you can do `PARSER=local yarn start`, `FIREBASE_DB=production yarn start`, etc.

File: src/common/constants.ts (hunk 1 above)
Comment (AlexNisnevich) [reply src/common/constants.ts:156]: Hmm, it's cause I fall back to staging, so we still have to cover all three cases.

File: webpack.config.js (hunk 3 above)
Comment (jacobnisnevich) [reply webpack.config.js:62]: Oh I see... I didn't fully understand how this plugin differed from the previous one

File: src/common/constants.ts (hunk 1 above)
Comment (jacobnisnevich) [reply src/common/constants.ts:156]: Ahh fair enough

**Review by jacobnisnevich — APPROVED** [review]
